lte/gateway/c/oai/tasks/amf/amf_config.c

```diff
@@ -141,7 +152,646 @@ void amf_config_init(amf_config_t* config) {
+
+    if (setting != NULL) {
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_OUTPUT, (const char**) &astring)) {
+        if (astring != NULL) {
+          if (config_aA->log_config.output) {
+            bassigncstr(config_aA->log_config.output, astring);
+          } else {
+            config_aA->log_config.output = bfromcstr(astring);
+          }
+        }
+      }
+
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_OUTPUT_THREAD_SAFE,
+              (const char**) &astring)) {
+        if (astring != NULL) {
+          config_aA->log_config.is_output_thread_safe = parse_bool(astring);
+        }
+      }
+
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_COLOR, (const char**) &astring)) {
+        if (strcasecmp("yes", astring) == 0)
+          config_aA->log_config.color = true;
+        else
+          config_aA->log_config.color = false;
+      }
+
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_SCTP_LOG_LEVEL,
+              (const char**) &astring))
+        config_aA->log_config.sctp_log_level = OAILOG_LEVEL_STR2INT(astring);
+
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_S1AP_LOG_LEVEL,
+              (const char**) &astring))
+        config_aA->log_config.s1ap_log_level = OAILOG_LEVEL_STR2INT(astring);
+
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_NAS_LOG_LEVEL,
+              (const char**) &astring))
+        config_aA->log_config.nas_log_level = OAILOG_LEVEL_STR2INT(astring);
+
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_MME_APP_LOG_LEVEL,
+              (const char**) &astring))
+        config_aA->log_config.mme_app_log_level = OAILOG_LEVEL_STR2INT(astring);
+
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_S6A_LOG_LEVEL,
+              (const char**) &astring))
+        config_aA->log_config.s6a_log_level = OAILOG_LEVEL_STR2INT(astring);
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_SECU_LOG_LEVEL,
+              (const char**) &astring))
+        config_aA->log_config.secu_log_level = OAILOG_LEVEL_STR2INT(astring);
+
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_UDP_LOG_LEVEL,
+              (const char**) &astring))
+        config_aA->log_config.udp_log_level = OAILOG_LEVEL_STR2INT(astring);
+
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_UTIL_LOG_LEVEL,
+              (const char**) &astring))
+        config_aA->log_config.util_log_level = OAILOG_LEVEL_STR2INT(astring);
+
+      if (config_setting_lookup_string(
+              setting, LOG_CONFIG_STRING_ITTI_LOG_LEVEL,
+              (const char**) &astring))
+        config_aA->log_config.itti_log_level = OAILOG_LEVEL_STR2INT(astring);
+#if EMBEDDED_SGW
```

Comment:
are we working with two different form factors of AGW? let us check with Ulas.

Reply:
MME code is shared between OAI core and Magma Core. EMBEDDED_SGW refers to Magma cases. Since 5G additions to MME will not be used by OAI, there should not be any conflicts on these and we can omit the configs not related to Magma 5G completely.

Reply:
Comments addressed.